vmware_dvs/agent/firewalls/vcenter_firewall.py

```diff
@@ -106,8 +113,12 @@ def _get_update_tasks(self):
     def _get_remove_tasks(self):
         while not self.remove_queue.empty():
             port = self.remove_queue.get()
-            self.removed[port['id']] = time.time()
-            self.remove_store.append(port)
+            dvs = self.get_dvs(port)
+            if dvs:
+                stored_tasks = self.remove_store.get(dvs, [])
```

Comment:
May be better use set() instead of list to avoid duplicate delete port tasks. But not sure about probability that it may happen